### DISCUSSION TITLE
Compressed glb file is not rendering in filament ( Android )

### DISCUSSION THREAD

**Discussion by udayk07** [google/filament#3815] [Q&A]
While trying to render compressed glb model in filament with sceneform, model is not displaying. I am using filament and gltfio v1.9.9. Found no specific logs in logcat. Can anyone help with this?

[c_stool.zip](https://github.com/google/filament/files/6326349/c_stool.zip)

**Comment by prideout** [comment]
This model works with today's gltf_viewer on a desktop machine.  However the WebGL viewer is reporting "memory access out of bounds", which is interesting.

![Screen Shot 2021-04-16 at 10 28 46 AM](https://user-images.githubusercontent.com/1288904/115062264-03763800-9e9f-11eb-9301-7afa431bd62a.png)

**Reply by prideout** [reply]
According to the following ASAN report, we have a bug in our tangents helper, I'll look into this.

```
==28346==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x00012cefa2c4 at pc 0x000107aec836 bp 0x7ffeedf85f30 sp 0x7ffeedf85f28
READ of size 4 at 0x00012cefa2c4 thread T0
    #0 0x107aec835 in filament::geometry::OrientationBuilderImpl::buildWithUvs() SurfaceOrientation.cpp:235
    #1 0x107ae7b76 in filament::geometry::SurfaceOrientation::Builder::build() SurfaceOrientation.cpp:141
    #2 0x1041d7098 in gltfio::TangentsJob::run(gltfio::TangentsJob::Params*) TangentsJob.cpp:158
    ```

**Reply by udayk07** [reply]
Thanks for the reply @prideout 

JFYI,
I tried using gltf modelviewer in Android app, model is not displayed and it is like screen flickering as in recording. While trying to use uncompressed version it works fine.
https://user-images.githubusercontent.com/78372917/115065089-584f9800-9f0b-11eb-9b61-e3f92577640e.mp4

**Reply by prideout** [reply]
The glTF accessor says a certain mesh has 85563 vertices, so that's what Filament's tangent generator expects.  However the decoded Draco mesh has 94473 vertices.  This leads to the heap overflow.

I'll see if I can make Filament more robust for this, but I suspect that this is a manifestation of a known issue with the Blender exporter:
https://github.com/KhronosGroup/glTF-Blender-IO/issues/1345
